test/integration/integration_test_fixtures.py

```diff
@@ -84,15 +90,22 @@ def setUp(self):
     """
     Performs sanity checks and starts up the server. Upon exit the server is ready to accept connections.
     """
-    assert (os.path.exists(self.nighthawk_test_server_path))
-    assert (os.path.exists(self.nighthawk_client_path))
-    test_id = os.environ.get('PYTEST_CURRENT_TEST').split(':')[-1].split(' ')[0].replace(
-        "[", "_").replace("]", "")
-    self.parameters["test_id"] = test_id
+    if os.getenv("NH_DOCKER_IMAGE", "") == "":
+      assert (os.path.exists(self.nighthawk_test_server_path))
+      assert (os.path.exists(self.nighthawk_client_path))
+
+    self.test_id = os.environ.get('PYTEST_CURRENT_TEST').split(':')[-1].split(' ')[0].replace(
+        "[", "_").replace("]", "").replace("/", "_")[5:]
+    self.tag = "{timestamp}/{test_id}".format(timestamp=TIMESTAMP, test_id=self.test_id)
+
     for i in range(self.backend_count):
-      test_server = NighthawkTestServer(self.nighthawk_test_server_path,
-                                        self.nighthawk_test_config_path, self.server_ip,
-                                        self.ip_version, self.parameters)
+      test_server = NighthawkTestServer(
```

Comment:
This setUp method is getting fairly long. Any chance we can extracts some logical sub-parts from it and move it into well named private functions (or methods if necessary) ?